File: lab7-mchib031/dataRestService-client/src/app/books/model/book.ts
export class Book {
  constructor(
    public id: number,
    public category: string,
    public title: string,
    public cost: number,
    public authors?: Author[],
    public year?: number,
    public description?: string
  ) {}
}

export class Author {
  constructor(
    public id: number,
    public firstName: string,
    public lastName: string
  ){}
}

File: lab7-mchib031/dataRestService-server/src/main/kotlin/seg3x02/booksrestapi/entities/Book.kt
package seg3x02.booksrestapi.entities

import javax.persistence.*

@Entity
class Book {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    var id: Long = 0
    var title: String = ""
    var category: String = ""
    var isbn: String = ""
    var cost: Double = 0.0
    var year: Int = 0
    var description: String = ""

    @ManyToMany(mappedBy = "books", cascade = [CascadeType.PERSIST, CascadeType.MERGE])
    var authors: MutableList<Author> = ArrayList()

    @OneToMany(cascade = [CascadeType.ALL])
    var orders: MutableList<Order> = ArrayList()
}
